packages/subgraph/src/crabV2.ts

```diff
@@ -0,0 +1,203 @@
+  userTx.wSqueethAmount = event.params.tradedAmountOut
+  userTx.ethAmount = (userTx.ethAmount !== null ? userTx.ethAmount : BigInt.fromString('0')).plus(event.transaction.value)
+  userTx.user = event.params.depositor
+  userTx.owner = event.transaction.from
+  userTx.type = 'FLASH_DEPOSIT'
+  userTx.timestamp = event.block.timestamp
+  userTx.save()
+}
+
+export function handleFlashWithdraw(event: FlashWithdraw): void {
+  const userTx = loadOrCreateTx(event.transaction.hash.toHex())
+  userTx.wSqueethAmount = event.params.wSqueethAmount
+  userTx.lpAmount = event.params.crabAmount
+  userTx.user = event.params.withdrawer
+  userTx.owner = event.transaction.from
+  userTx.type = 'FLASH_WITHDRAW'
+  userTx.timestamp = event.block.timestamp
+  userTx.save()
+}
+
+export function handleFlashDepositCallback(event: FlashDepositCallback): void {
+  const userTx = loadOrCreateTx(event.transaction.hash.toHex())
+  userTx.ethAmount = ((userTx.ethAmount !== null ? userTx.ethAmount : BigInt.fromString('0')) as BigInt).minus(event.params.excess)
+  userTx.type = 'FLASH_DEPOSIT_CALLBACK'
+  userTx.save()
+}
+
+export function handleFlashWithdrawCallback(event: FlashWithdrawCallback): void {
+  const userTx = loadOrCreateTx(event.transaction.hash.toHex())
+  userTx.ethAmount = event.params.excess
+  userTx.type = 'FLASH_WITHDRAW_CALLBACK'
+  userTx.save()
+}
+
+export function handleSetHedgeTimeThreshold(event: SetHedgeTimeThreshold): void {
+  const timeThreshold = new CrabHedgeTimeThreshold(event.transaction.hash.toHex())
+  timeThreshold.threshold = event.params.newHedgeTimeThreshold;
+  timeThreshold.timestamp = event.block.timestamp;
+  timeThreshold.save()
+}
+
+export function handleQueueTransaction(event: QueueTransaction): void {
+  const tx = new TimeLockTx(event.params.txHash.toHex());
+  tx.target = event.params.target;
+  tx.value = event.params.value;
+  tx.signature = event.params.signature;
+  tx.data = event.params.data;
+  tx.eta = event.params.eta
+  tx.queued = true;
+  tx.timestamp = event.block.timestamp;
+  tx.save()
+}
+
+export function handleExecuteTransaction(event : QueueTransaction): void {
+  const execTimeLockTx = new ExecuteTimeLockTx(event.params.txHash.toHex());
+  execTimeLockTx.timestamp = event.block.timestamp;
+  const id = event.params.txHash.toHex();
+  const tx = TimeLockTx.load(id);
+  if(tx) {
+    tx.queued = false
+    tx.save();
+    execTimeLockTx.timelocktx = tx.id;
+  }
+  execTimeLockTx.save();
+}
+
+export function handleHedgeOTC(event: HedgeOTC): void {
+  const hedge = new HedgeOTCSchema(event.params.bidId.toString());
+  hedge.bidID = event.params.bidId;
+  hedge.clearingPrice = event.params.clearingPrice;
+  hedge.quantity = event.params.quantity;
+  hedge.isBuying = event.params.isBuying;
+  hedge.timestamp = event.block.timestamp;
+  hedge.save();
+}
+
+export function handleHedgeOTCSingle(event: HedgeOTCSingle): void {
+  const hedge = new HedgeOTCSingleSchema(event.params.bidId.toString());
```

Comment:
Same! Also, every event for the same tx will have the same bid id and tx. So we need something more. I would recommend using tx-logId

Reply:
This done